File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        
        some = sum(nums[:k])
        maxi = some/k
        for i in range(k,len(nums)):
            some+=nums[i] - nums[i-k]
            avg = some/k
            maxi = max(maxi,avg)
        return maxi

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        
        double sum=0;
        double avg = 0;
        double maxAvg = Integer.MIN_VALUE;

        // 1st window
        for(int i=0;i<k;i++){
            sum += nums[i];
        }
        avg = sum/k;
        maxAvg = Math.max(avg, maxAvg);

        for(int j=k;j<nums.length;j++){
            sum -= nums[j-k];
            sum += nums[j];
            avg = sum/k;
            maxAvg = Math.max(avg, maxAvg);

        }

        return maxAvg;

    }
}